languages/eo.lua

```diff
@@ -1,4 +1,87 @@
+}
+
+SILE.settings:declare({ parameter = "languages.eo.use_raised_ordinal_suffix", type = "boolean", default = false })
+SILE.settings:declare({ parameter = "languages.eo.hyphen_before_ordinal_suffix", type = "boolean", default = false })
+
+SILE.doTexlike([[%
+\define[command=tableofcontents:title:eo]{Enhavoj}%
+\define[command=book:chapter:pre:eo]{Ĉapitro }%
+]])
+
+local function eo_nth(n)
+  if n == nil then
+    n = ""
+  end
+  local a
+  a = SILE.settings:get("languages.eo.use_raised_ordinal_suffix") and "ª" or "a"
+  if SILE.settings:get("languages.eo.hyphen_before_ordinal_suffix") then
+    a = "-" .. a
+  end
+  return n .. a
+end
+
+SILE.registerCommand("book:chapter:post:eo", function()
+  SILE.typesetter:typeset(eo_nth())
+  SILE.call("medskip", {})
```

Comment:
Typesetting decisions in a language-specific file seem wrong to me. We'd perhaps need another discussion about these ugly hooks), still something looks amiss (Why a medskip? That should be a class decision, no?)

Reply:
https://github.com/sile-typesetter/sile/blob/b211d9d842276f66b88e2107aea52249fd7c63d1/languages/ja.lua#L183

Reply:
https://github.com/sile-typesetter/sile/blob/b211d9d842276f66b88e2107aea52249fd7c63d1/languages/fr.lua#L25-L51

Reply:
I've no idea if the first mention for japanese is correct, and no reason to believe it....

The second mention regarding the French stuff is wholly unrelated: that's typography according to the rules, not decisions how chapter numbers could be presented.) Also note that they are customizable (as Canadian and Swiss, for instance, have some variations of these.)

Unless there are normative references for the presentation of chapters, that say a medskip (with a given definition) is the only typography-respecting way for this, it's at best an author decision...

Reply:
dude whatever

Reply:
> dude whatever

?

Sorry I don't get your reply.

Maybe I have just misunderstood the very idea of introducing the "fluent" things, but with its, the **book** class now defines by default:
```
  SILE.registerCommand("book:chapter:post", function (_, _)
    SILE.call("fluent", {}, { "book-chapter-post" })
    SILE.call("par")
  end)
```

It's true that `\book:sectioning` in that class still has the existence check on a command `\book:chapter:post:xxx` (where xxx is the language), but only the Japanese language file has it. It seems to me we should deprecate it there.

The _only_ difference the "ja" (and now the "eo") have is a medskip instead of a par. It doesn't seem logical -- spacing between a "book" chapter header and its title is something that should only depend on the so-called "book" class, no? That's the only thing I am saying: skips are a class-decision, not a translation / language support issue.

Reply:
The `\medskip` in the *j* classes/languages is leftover from a very early stage of SILE development.

At this point language support files shouldn't not be registering commands of any sort unless absolutely necessary, and the only thing that needs changing here is the `book-chapter-post` key in the Fluent sources. It looks like that may be incoming after this commit anyway...